### PR TITLE
Document how to set up custom WDA server

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
## Proposed changes

Added documentation on how to set up and configure custom WDA server for appium xcuitest driver. This strategy should, hopefully, help to resolve known XCTest/WDA issues like https://github.com/facebook/WebDriverAgent/issues/507 and speed up automated tests initialization in general

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Comment by triager** [comment]
Can one of the admins verify this patch?

**Comment by triager** [comment]
Can one of the admins verify this patch?

File: docs/en/advanced-concepts/wda-custom-server.md [hunk 1]
```diff
@@ -0,0 +1,313 @@
+## How To Set Up And Customize WebDriverAgent Server
+
+Appium for iOS uses [Facebook's WebDriverAgent](https://github.com/facebook/WebDriverAgent)
+as automation backend. This backend is based on Apple's XCTest framework and shares all the
```
Comment (jlipps) [inline docs/en/advanced-concepts/wda-custom-server.md]: `as (the|an|its) automation backend`

File: docs/en/advanced-concepts/wda-custom-server.md [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+known problem that are present in XCTest. For some of them we have workarounds, but there
+are some, that are hardly possible to workaround. The approach described in this article
```
Comment (jlipps) [inline docs/en/advanced-concepts/wda-custom-server.md]: `some that are hardly possible to work around`

**Review by jlipps — COMMENTED** [review]
This is fantastic and thorough. I made a couple of grammatical suggestions. Also, it wasn't clear to me on a first read _why_ I might want to run WDA on my own. Perhaps you could add a note as to why? For example:

> For some of them we have workarounds, but there are some that are hardly possible to workaround. The approach described in this article enables you to have full control over how WDA is built, managed, and run on the device. This way, if you have custom WDA extensions, or want to change the way WDA processes are managed to tune it to your CI environment, you are able to do so.

File: docs/en/advanced-concepts/wda-custom-server.md (hunk 1 above)
Comment (mykola-mokhnach) [inline docs/en/advanced-concepts/wda-custom-server.md]: fixed

File: docs/en/advanced-concepts/wda-custom-server.md (hunk 2 above)
Comment (mykola-mokhnach) [inline docs/en/advanced-concepts/wda-custom-server.md]: fixed

**Comment by mykola-mokhnach** [comment]
@jlipps your comments have been addressed. Please check the PR once more.

**Review by jlipps — APPROVED** [review]

**Review by imurchie — APPROVED** [review]

**Pull request merged**